packages/synapse-react-client/src/components/ChallengeSubmission/ChallengeSubmissionStepper.tsx

```diff
@@ -0,0 +1,206 @@
+  const steps = getSteps(entityType)
+  const initialStep =
+    entityType === EntityType.DOCKER_REPO
+      ? steps.SELECT_COMMIT
+      : steps.SELECT_EVALUATION
+  const [step, setStep] = useState<Step>(initialStep)
+  const [errorMessage, setErrorMessage] = useState<string>()
+  const [selectedCommit, setSelectedCommit] = useState<DockerCommit>()
+
+  const [submissionName, setSubmissionName] = useState<string>('')
+  const [selectedEval, setSelectedEval] = useState<string | undefined>()
+  const [submissionError, setSubmissionError] = useState<string | undefined>()
+  const [confirming, setConfirming] = useState<boolean>(false)
+  const [loading, setLoading] = useState<boolean>(false)
+
+  const submitRepoForEvaluation = async () => {
+    if (!entity.id || !selectedCommit) return
+
+    const submission: EvaluationSubmission = {
+      userId: userId,
+      evaluationId: selectedEval!,
+      entityId: entity.id,
+      dockerRepositoryName: entity.repositoryName,
+      dockerDigest: selectedCommit.digest,
+      versionNumber: 1,
+      teamId: teamId,
+    }
+
+    if (submissionName !== '') submission['name'] = submissionName
+    try {
+      await SynapseClient.submitToEvaluation(
+        submission,
+        entity.etag!,
+        accessToken,
+      )
+    } catch (e) {
+      setErrorMessage(e.message)
+    }
+  }
+
+  const hide = () => {
+    setErrorMessage(undefined)
+    setSelectedCommit(undefined)
+    setSubmissionName('')
+    setSelectedEval(undefined)
+    setSubmissionError(undefined)
+    setConfirming(false)
+    onClose()
+    setStep(initialStep)
+  }
+
+  function handleStepChange(value?: StepsEnum) {
+    console.log({ handleStepChange: value })
+    if (!value || !steps[value]) return
+    setErrorMessage(undefined)
+    setStep(steps[value])
+  }
+
+  const onConfirmHandler = async () => {
+    console.log('onConfirm')
+    await submitRepoForEvaluation()
+  }
+
+  const onCommitChanged = (value: DockerCommit) => {
+    console.log(value)
+    setSelectedCommit(value)
+    setStep({ ...step, nextEnabled: true })
+  }
+
+  function stepperContent() {
+    switch (step.id) {
+      case StepsEnum.SELECT_COMMIT:
+        return (
+          <SubmissionCommitList
+            repository={entity}
+            selectedCommit={selectedCommit}
+            onCommitChanged={onCommitChanged}
+          />
+        )
+        break
+      case StepsEnum.SELECT_EVALUATION:
+        return (
+          <EvaluationQueueList
+            projectId={projectId}
+            submissonName={submissionName}
+            onSubmissionNameChange={setSubmissionName}
+            selectedEvaluation={selectedEval}
+            onEvaluationChange={value => {
+              setStep({ ...step, confirmEnabled: true })
+              setSelectedEval(value)
+            }}
+          />
+        )
+      case StepsEnum.SUBMISSION_SUCCESS:
+        return (
+          <Typography variant="body1" sx={{ fontSize: '16px' }}>
```

Comment:
Is this really 'body2' (if the fontSize needs to be adjusted)?

Reply:
I agree it's not. There are several font-size mismatches between the available variants and the styles called for in Figma. How would it be best to handle these cases?

Reply:
Your override is fine for this PR.  This might be a new variant from design, so web engineering should follow up.